plugins/decklink/decklink-output.cpp

```diff
@@ -32,13 +32,30 @@ static void *decklink_output_create(obs_data_t *settings, obs_output_t *output)
 		DeckLinkDeviceMode *mode =
 			device->FindOutputMode(decklinkOutput->modeID);
 
+		ComPtr<IDeckLinkOutput> output_;
+		device->GetOutput(&output_);
+
+		BMDDisplayMode *actualMode = nullptr;
+		decklink_bool_t modeSupportsBGRA = false;
+		output_->DoesSupportVideoMode(bmdVideoConnectionUnspecified,
+					      mode->GetDisplayMode(),
+					      bmdFormat8BitBGRA,
+					      bmdNoVideoOutputConversion,
+					      bmdSupportedVideoModeKeying,
+					      actualMode, &modeSupportsBGRA);
+
+		const bool supportsHDR = (device->GetSupportsHDRMetadata() &&
+					  !decklinkOutput->force_sdr);
+
 		struct video_scale_info to = {};
-		to.format = VIDEO_FORMAT_BGRA;
+		to.format = (modeSupportsBGRA && supportsHDR)
+				    ? VIDEO_FORMAT_BGRA
+				    : VIDEO_FORMAT_UYVY;
```

Comment:
I would prefer not dropping down to UYVY unless the keyer was both unsupported and requested by user setting. That probably means reacting to KEYER setting updates in decklink_output_update among other changes.

Reply:
The keyer flag is a red herring here, I'll go ahead and remove it or clarity. These video modes don't support BGRA without the keyer either which is the real issue.

Reply:
I think we should call `DoesSupportVideoMode` with `keyerMode ? bmdSupportedVideoModeKeying : bmdSupportedVideoModeDefault` or something along those lines to be sure.

Reply:
Well the keyer requires the RGBA mode to work so just checking for RGBA should cover both. We may want to grey out the keyer options per video mode in the UI with that check though.

Reply:
Ok, I'm happy with Default. Thanks.